### PR TITLE
update the desktop & appstream xml for Linux

### PR DESCRIPTION
**Pull request by shenlebantongying**
related https://github.com/xiaoyifang/goldendict-ng/issues/690

* rename to -ng
* add Chinese translations
* add keywords

kde display GenericName
image

while xfce displays Comment
image

---

appstream XML is used in "software stores" on linux like

https://flathub.org/apps/org.goldendict.GoldenDict
https://appstream.ubuntu.com/lunar/universe/metainfo/goldendict-webengine.html

Changing the id should prepare everything needed if someone wants to make a Flatpak package.

The XML file is checked via

```
appstream-util validate-relax ./org.xiaoyifang.GoldenDict_NG.metainfo.xml
```

### CODE REVIEW

**Comment by xiaoyifang** [comment]
`_NG`  or `-ng`  or `_ng`?

File: redist/org.xiaoyifang.GoldenDict_NG.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2019 Vitaly Zaitsev <vitaly@easycoding.org> -->
+<component type="desktop">
+  <id>org.xiaoyifang.GoldenDict_NG</id>
```
Comment (xiaoyifang) [inline redist/org.xiaoyifang.GoldenDict_NG.metainfo.xml:4]: org.GoldenDict.ng?

Comment (shenlebantongying) [reply redist/org.xiaoyifang.GoldenDict_NG.metainfo.xml:4]: According to https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic

> The ID must follow a reverse-DNS scheme, consisting of {tld}.{vendor}.{product}

**Comment by shenlebantongying** [comment]
> _NG or -ng or _ng?

Display as `GoldenDict-NG` in user-facing places like `.desktop` and use `goldendict_ng` in other places because the hyphen (-) seems to have trouble in some places.

https://www.freedesktop.org/software/appstream/metainfocreator/#/guiapp

image

**Comment by xiaoyifang** [comment]
what about this `com.goldendict.ng.desktop`?

File: redist/org.xiaoyifang.GoldenDict_NG.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    <screenshot>
+      <image>https://xiaoyifang.github.io/goldendict-ng/img/windows_white.webp</image>
+    </screenshot>
+  </screenshots>
+  <url type="homepage">https://github.com/xiaoyifang/goldendict-ng</url>
+  <update_contact>https://github.com/xiaoyifang/goldendict-ng</update_contact>
+  <launchable type="desktop-id">org.xiaoyifang.GoldenDict_NG.desktop</launchable>
+  <content_rating type="oars-1.0" />
+  <provides>
+    <id>org.goldendict_ng.desktop</id>
```
Comment (xiaoyifang) [inline redist/org.xiaoyifang.GoldenDict_NG.metainfo.xml:33]: should this id also be changed ?

Comment (shenlebantongying) [reply redist/org.xiaoyifang.GoldenDict_NG.metainfo.xml:33]: The original `privdes->id` is just this:
image

---

https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-provides

This id is used for other software that uses `<require>` tag to describe that they depend on another software that `<provides>`. Others can also `<suggest>` an id that we `<provides>`

Just append a _ng to the original should be OK.

(However, I failed to find who actually uses the `<require>` or `<suggest>` tag :sweat_smile: )

---

Most GUI apps don't even have `provides -> id` (like qbittorrent, okular....). 

https://appstream.ubuntu.com/focal/universe/metainfo/index.html

Comment (xiaoyifang) [reply redist/org.xiaoyifang.GoldenDict_NG.metainfo.xml:33]: ok

**Pull request merged**